src/theme/searcher/mod.rs

```diff
@@ -0,0 +1,56 @@
+//! Theme dependencies for in-browser search. Not included in mdbook when
+//! the "search" cargo feature is disabled.
+
+use std::path::Path;
+
+use theme::load_file_contents;
+
+pub static JS: &'static [u8] = include_bytes!("searcher.js");
+pub static MARK_JS: &'static [u8] = include_bytes!("mark.min.js");
+pub static ELASTICLUNR_JS: &'static [u8] = include_bytes!("elasticlunr.min.js");
+
+/// The `Searcher` struct should be used instead of the static variables because
+/// the `new()` method will look if the user has a searcher directory in their source folder and use
+/// the user's searcher instead of the default.
+///
+/// You should only use the static variables if you need the default searcher even if the
+/// user has specified another searcher.
+pub struct Searcher {
+    pub js: Vec<u8>,
+    pub mark_js: Vec<u8>,
+    pub elasticlunr_js: Vec<u8>,
+}
+
+impl Searcher {
+    pub fn new(src: &Path) -> Self {
```

Comment:
Should we create a test for this to make sure the file loading logic is sound and people are able to override the scripts?